base/compiler/ssair/passes.jl

```diff
@@ -720,6 +720,97 @@ function perform_lifting!(compact::IncrementalCompact,
     return stmt_val # N.B. should never happen
 end
 
+function lift_svec_ref!(compact::IncrementalCompact, idx::Int, stmt::Expr)
```

Comment:
This should probably be removed and replaced with an inference `tfunc` for `_svec_ref`, since it is rather unfortunate (generally) when optimization ends up with a more precise answer than inference; it tends to confuse and annoy people, since it is usually less reliable and accurate.

Reply:
This lifts `svec_ref(_compute_sparams(...), ...)` which is inserted by inlining and does not exist at inference time.

Reply:
It seems possible that the optimizer could have already lifted that static_parameter expr into a `typeof` call inside the method, before it reached here. Is the optimizer not accurately marking the `svec_ref` with the result type constant (if known from inference) when it inserts it? Still seems confusing that inference couldn't determine this static_parameter value when the IR was simple, but optimization could figure it out after the IR was made more complicated.

Reply:
If the optimizer knows the type of the static parameter, it shouldn't insert the svec_ref (it currently does on master, but that's a bug fixed in https://github.com/JuliaLang/julia/pull/46703). Note that the optimizer here is not figuring out the *value* of the static parameters, but merely an SSAValue that will contain that value at runtime, which is a different analysis.

Reply:
Ah, okay, I thought this was attempting to forward the type. It makes more sense that this is attempting to lift the computation of `T` through a call that just gets the parameter back directly, e.g. `eltype(A{T}) -> T`